File: app/src/main/java/com/d4rk/androidtutorials/java/ui/android/buttons/radio/RadioButtonsActivity.java
package com.d4rk.androidtutorials.java.ui.android.buttons.radio;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;
import android.widget.RadioButton;

import androidx.appcompat.app.AppCompatActivity;

import com.d4rk.androidtutorials.java.databinding.ActivityRadioButtonsBinding;
import com.google.android.material.snackbar.Snackbar;

public class RadioButtonsActivity extends AppCompatActivity {
    private final Handler handler = new Handler();
    private ActivityRadioButtonsBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityRadioButtonsBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());
        binding.floatingButtonShowSyntax.setOnClickListener(v -> startActivity(new Intent(this, RadioButtonsCodeActivity.class)));
        binding.buttonDisplay.setOnClickListener(v -> {
            int selectedId = binding.radioGroup.getCheckedRadioButtonId();
            RadioButton radioButton = findViewById(selectedId);
            Snackbar.make(binding.getRoot(), radioButton.getText(), Snackbar.LENGTH_SHORT).show();
        });
        handler.postDelayed(() -> binding.floatingButtonShowSyntax.shrink(), 5000);
    }
}

File: app/src/main/kotlin/com/d4rk/androidtutorials/java/ui/android/buttons/radio/RadioButtonsActivity.kt
package com.d4rk.androidtutorials.java.ui.android.buttons.radio
import android.content.Intent
import android.os.Bundle
import android.os.Handler
import android.widget.RadioButton
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.d4rk.androidtutorials.java.databinding.ActivityRadioButtonsBinding
class RadioButtonsActivity : AppCompatActivity() {
    private lateinit var binding: ActivityRadioButtonsBinding
    @Suppress("DEPRECATION")
    private val handler = Handler()
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityRadioButtonsBinding.inflate(layoutInflater)
        setContentView(binding.root)
        binding.floatingButtonShowSyntax.setOnClickListener {
            startActivity(Intent(this, RadioButtonsCodeActivity::class.java))
        }
        binding.buttonDisplay.setOnClickListener {
            val selectedId = binding.radioGroup.checkedRadioButtonId
            val radioButton = findViewById<RadioButton>(selectedId)
            Toast.makeText(this, radioButton.text, Toast.LENGTH_SHORT).show()
        }
        handler.postDelayed({
            binding.floatingButtonShowSyntax.shrink()
        }, 5000)
    }
}